sklearn/metrics/_scorer.py

```diff
@@ -261,7 +261,9 @@ def __repr__(self):
             f"{response_method_string}{kwargs_string})"
         )
 
-    def __call__(self, estimator, X, y_true, sample_weight=None, **kwargs):
+    # TODO (1.9): remove in 1.9
+    @_deprecate_positional_args(version="1.9")
+    def __call__(self, estimator, X, y_true, *, sample_weight=None, **kwargs):
```

Comment:
Out of curiosity, why do we need to deprecate `sample_weight` as a positional arg ? Is it directly related to this PR or is it a general guideline ?